### PR TITLE
Adding Windows Support - Refactored code to fix a few compilation issues...

### PR DESCRIPTION
**Pull request by sriniprash**
* Incorporated the comments given for the previous PULL Request.
* The changes are compiling & build is working fine in windows.

### CODE REVIEW

**Comment by kellegous** [comment]
@sriniprash I'm going to take a look a this, I just have to bring my windows VM back up so I can try it out.

File: src/code.google.com/p/codesearch/index/mmap_windows.go [hunk 1]
```diff
@@ -11,6 +11,29 @@ import (
...
+	// For now, only the calls from read.go are handled using the windows equivalent of Munmap
+	// i.e syscall.UnmapViewOfFile(o uintptr)
+	// Calls from write.go get a nil always.
+	// TODO: Find a way to get the mmapData.o field in write.go.
+
+	switch o.(type) {
+		case uintptr:
+			return syscall.UnmapViewOfFile(o.(uintptr))
+		default:
+			return nil
```
Comment (kellegous) [inline src/code.google.com/p/codesearch/index/mmap_windows.go]: Do I read this correctly? On Windows, we would not be unmapping the regions associated with the index writers?

Comment (kellegous) [inline src/code.google.com/p/codesearch/index/mmap_windows.go]: I think the right answer here is to have the index writers also hold a proper reference to the whole mmapData and not just the slice taken from the mmap region. Maybe it's time to have mmapData be an io.Closer so we don't have to keep up with what field needs to be passed to Munmap.

File: src/code.google.com/p/codesearch/index/mmap_bsd.go [hunk 2]
```diff
@@ -12,6 +12,17 @@ import (
   "syscall"
 )
 
+// An mmapData is mmap'ed read-only data from a file.
```
Comment (kellegous) [inline src/code.google.com/p/codesearch/index/mmap_bsd.go]: We might be able to move the _bsd and _linux code into a single mmap_unix.go file.

File: src/code.google.com/p/codesearch/index/mmap_windows.go (hunk 1 above)
Comment (sriniprash) [inline src/code.google.com/p/codesearch/index/mmap_windows.go]: Yea. That's right. Currently it was not possible to unmap the regions in windows due to lack of enough info in mmapData to do so.

File: src/code.google.com/p/codesearch/index/mmap_bsd.go (hunk 2 above)
Comment (sriniprash) [inline src/code.google.com/p/codesearch/index/mmap_bsd.go]: Yea. This should be possible.

File: src/code.google.com/p/codesearch/index/mmap_windows.go (hunk 1 above)
Comment (sriniprash) [inline src/code.google.com/p/codesearch/index/mmap_windows.go]: The irst approach (holding mmapData reference) looks doable. Could you please elaborate on the second one?

**Comment by ameerkat** [comment]
Does this build completely? I had to modify the the ansi/tty files as well on my machine.

**Pull request closed without merging**